scripts/zarr_s3_poc.R

```diff
@@ -0,0 +1,66 @@
+# This script explores three different ways to read Zarr files in R from an S3 bucket.
+
+# Option 1: Using stars package
+# from https://www.r-bloggers.com/2022/09/reading-zarr-files-with-r-package-stars/
+install.packages("sf")
+install.packages("stars")
+library(stars)
+# dsn <- 'ZARR:"/vsicurl/https://ncsa.osn.xsede.org/Pangeo/pangeo-forge/gpcp-feedstock/gpcp.zarr"'
+bounds <- c(longitude = "lon_bounds", latitude = "lat_bounds")
+dsn <- 'ZARR:"/vsis3/cmip6-pds/CMIP3/CCCMA/cccma_cgcm3_1/1pctCO2/r1i1p1f1/Amon/pr/lat/"'
+r <- read_mdim(dsn, bounds = bounds)
+# results in:
+# Error: file not found
+# In addition: Warning messages:
+# 1: In CPL_read_mdim(file, array_name, options, o
+
+# From online chatter this appears to be a Mac specific issue and related to GDAL being compiled without blosc compression support.
+
+
+# Option 2: Using reticulate
+install.packages("reticulate")
+library(reticulate)
+# By default a r-reticulate virtualenv is created, others can be created with virtualenv_create()
+virtualenv_list()
+env <- "r-reticulate"
+use_virtualenv(env)
+virtualenv_install(env, "zarr")
+virtualenv_install(env, "s3fs")
+
```

Comment:
@carlosgjs 
Since we are using `reticulate` to interface with Python, couldn't we just utilize Xarray or execute a Python script that handles all the slicing and loading? If so, we could use Option 2.

Reply:
@uwcdc 
See some of the comments below. One is that the reticulate interop introduces some friction, e.g. using the generator returned from `grp$array_keys()`. In general I expect some data types will require a bit of additional handling. The other issue is that at least on my setup this seems slower than the `Rarr`. I can do some benchmarking around this.

Have you tried using xarray with reticulate? If so, how's the interfacing and perf?